### PR TITLE
Using Gem::Version for semantic version checking

### PR DESCRIPTION
**Pull request by ennukee**
title

addresses #22

### CODE REVIEW

**Comment by dfaulken** [comment]
I know you didn't cause it, but any thoughts how to fix the Travis build failure?

**Comment by ennukee** [comment]
Not sure, but I am having issues running migrations which I put in slack (maybe part of the issue?)

**Comment by Anbranin** [comment]
I know what caused the travis build to fail, and I fixed it in my latest branch. It was just running two migrations with the same name, so I deleted one of them. Dummy app problems.

**Comment by ennukee** [comment]
[I'm not sure why we're telling the generator to make a migration](https://github.com/umts/microservices-engine/pull/46/files#diff-1dabf90ea18d6e2a125983155a92f688L10) when the engine itself [already has a migration](https://github.com/umts/microservices-engine/blob/master/db/migrate/20160914202818_create_microservices_engine_connections.rb) in `/db/migrate`. I mentioned it briefly on slack but you should still be able to access the connection objects in the dummy app without it having its own migration, so we might as well save the space and leave the excess logic, unless it works differently for actual apps that just implement the gem in the gemspec? 

It seems the intended functionality is that the master app will run all migrations in the engine's /db/migrate as well as its own. Below is a picture in the dummy app's `rails c` with solely the migration in /db/migrate.

screen shot 2017-02-06 at 1 32 59 pm

You can also see that travis is running the migrations as well.

screen shot 2017-02-06 at 1 37 26 pm

**Comment by Anbranin** [comment]
Hey @enragednuke, I don't know about the dummy app, but I tried importing the engine into both the round3 branch and the NTD-surveys app that David and I have running on our digital ocean droplet. I ran migrations, and the migration in the Engine was not run. Am I missing something? Maybe we could go over it together at some point, because of course it would be much better to not have this line.

**Comment by ennukee** [comment]
I did mention I haven't tested it w/ external apps (only with the dummy app), so it's possible it does matter in those cases.

File: lib/generators/install/install_generator.rb [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
-class InstallGenerator < ActiveRecord::Generators::Base 
+class InstallGenerator < ActiveRecord::Generators::Base
   source_root File.expand_path('../templates', __FILE__)
 
   def setup_initializer
     template 'mse_router_info.yml', 'config/mse_router_info.yml'
     copy_file 'microservices_engine.rb',
               'config/initializers/microservices_engine.rb'
-    generate 'migration', 'CreateMicroservicesEngineConnection name:string object:string url:string'
+    # generate 'migration', 'CreateMicroservicesEngineConnection name:string object:string url:string'
```
Comment (Anbranin) [inline lib/generators/install/install_generator.rb:10]: approving knowing we might have to use this line anyways.. but we'll revisit it.

**Review by Anbranin — APPROVED** [review]

**Pull request merged**